modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java

```diff
@@ -0,0 +1,132 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.commerce.delivery.catalog.internal.resource.v1_0;
+
+import com.liferay.commerce.product.model.CommerceChannel;
+import com.liferay.commerce.product.service.CommerceChannelService;
+import com.liferay.commerce.wish.list.model.CommerceWishList;
+import com.liferay.commerce.wish.list.service.CommerceWishListService;
+import com.liferay.headless.commerce.core.util.ServiceContextHelper;
+import com.liferay.headless.commerce.delivery.catalog.dto.v1_0.WishList;
+import com.liferay.headless.commerce.delivery.catalog.internal.dto.v1_0.converter.WishListDTOConverter;
+import com.liferay.headless.commerce.delivery.catalog.resource.v1_0.WishListResource;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Mahmoud Azzam
+ */
+@Component(
+	properties = "OSGI-INF/liferay/rest/v1_0/wish-list.properties",
+	scope = ServiceScope.PROTOTYPE, service = WishListResource.class
+)
+public class WishListResourceImpl extends BaseWishListResourceImpl {
+
+	@Override
+	public Response deleteWishList(Long wishListId) throws Exception {
+		_commerceWishListService.deleteCommerceWishList(wishListId);
+
+		Response.ResponseBuilder responseBuilder = Response.noContent();
+
+		return responseBuilder.build();
+	}
+
+	@Override
+	public Page<WishList> getChannelWishListsPage(
+			Long channelId, Long accountId, Pagination pagination)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelService.fetchCommerceChannel(channelId);
+
+		return Page.of(
+			transform(
+				_commerceWishListService.getCommerceWishLists(
+					commerceChannel.getSiteGroupId(), contextUser.getUserId(),
+					pagination.getStartPosition(), pagination.getEndPosition(),
+					null),
+				commerceWishList -> _toWishList(commerceWishList)),
+			pagination,
+			_commerceWishListService.getCommerceWishListsCount(
+				commerceChannel.getSiteGroupId(), contextUser.getUserId()));
+	}
+
+	@Override
+	public WishList getWishList(Long channelId, Long accountId, Long wishListId)
+		throws Exception {
+
+		return _toWishList(
+			_commerceWishListService.getCommerceWishList(wishListId));
+	}
+
+	@Override
+	public WishList patchChannelWishList(
+			Long channelId, Long accountId, Long wishListId, WishList wishList)
+		throws Exception {
+
+		return _toWishList(
+			_commerceWishListService.updateCommerceWishList(
+				wishListId, wishList.getName(), wishList.getDefaultWishList()));
```

Comment:
use getterutil and the defualt value will be the already saved value for both name and default wishlist

Reply:
also you're not using channelid and accountid so remove parameters

Reply:
so also the endpoint is wrong